x-pack/plugins/security_solution/common/transforms/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+/**
+ * Kibana configuration schema
+ */
+export const transformConfigSchema = schema.object({
+  auto_start: schema.boolean(),
+  auto_create: schema.boolean(),
+  enabled: schema.boolean(),
+  query: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+  retention_policy: schema.maybe(
+    schema.object({
+      time: schema.object({
+        field: schema.string(),
+        max_age: schema.string(),
+      }),
+    })
+  ),
```

Comment:
Thoughts on there being a default retention policy here? Even if it's super long, seems best to at least have a fallback?

Reply:
Yeah, we can. What is super long? Should it be 1 week? 1 year?

Reply:
I think a week seems like a reasonable default (reaching into the air on this one 😅). Of course this will depend deployment-by-deployment, but being able to select `last week` and still leverage the entity/summary tables seems like a good tradeoff with storage. Will want to change/revisit once we have more data around usage.

Reply:
Looking further, I have a default of 3d for this one, this is a maybe where it allows the end user to remove it from the settings and then no retention policy will be used. But out of the box examples and default will be 1w.

I don't know want to force one when the underlying transform API allows there to not be one. So I will fix the 3d -> 1w, but leave this here as a maybe.